WireAPI/Tests/WireAPITests/UpdateEventsAPITests.swift

```diff
@@ -0,0 +1,97 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import SnapshotTesting
+@testable import WireAPI
+import XCTest
+
+final class UpdateEventsAPITests: XCTestCase {
+
+    // MARK: - Request generation
+
+    func testGetLastUpdateEvent() async throws {
+        let snapshotter = APISnapshotHelper { httpClient, apiVersion in
+            UpdateEventsAPIBuilder(httpClient: httpClient)
+                .makeAPI(for: apiVersion)
+        }
+
+        try await snapshotter.verifyRequestForAllAPIVersions { sut in
+            _ = try await sut.getLastUpdateEvent(selfClientID: Scaffolding.selfClientID)
+        }
+    }
+
+    // MARK: - Response handling
+
+    // MARK: - V0
+
+    func testGetLastUpdateEvent_200_V0() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(
+            code: 200,
+            payloadResourceName: "GetLastEventSuccessResponseV0"
+        )
+
+        let sut = UpdateEventsAPIV0(httpClient: httpClient)
+
+        // When
+        let result = try await sut.getLastUpdateEvent(selfClientID: Scaffolding.selfClientID)
+
+        // Then
+        XCTAssertEqual(result, Scaffolding.updateEventEnvelope)
+    }
+
+    func testGetLastUpdateEvent_400_V0() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(code: 400, errorLabel: "")
+        let sut = UpdateEventsAPIV0(httpClient: httpClient)
+
+        // Then
+        await assertAPIError(UpdateEventsAPIError.invalidClient) {
+            // When
+            _ = try await sut.getLastUpdateEvent(selfClientID: Scaffolding.selfClientID)
+        }
+    }
+
+    func testGetLastUpdateEvent_404_V0() async throws {
+        // Given
+        let httpClient = try HTTPClientMock(code: 404, errorLabel: "not-found")
+        let sut = UpdateEventsAPIV0(httpClient: httpClient)
+
+        // Then
+        await assertAPIError(UpdateEventsAPIError.notFound) {
```

Comment:
Should we stick to the Apple api `XCTAssertThrowsError` to make it easy to understand for new developers?
Maybe `AssertThrowsAPIError` works?

Reply:
And then also the way the parameters are passed is slightly different.. one example from our code base
```swift
        XCTAssertThrowsError(try handleChangeState.validate("TestUser")) { error in
            XCTAssertEqual(error as! HandleChangeState.ValidationError,
                                        HandleChangeState.ValidationError.invalidCharacter)
        }
```

Reply:
There's no `XCTAssetThrowsError` variant that supports async expressions, but I'll add one.

Reply:
done 92ec8c569ff5a834a2a75fea2e147e2ed9fb8f51

Reply:
Thank you!